api/src/test/java/org/openmrs/api/ConceptServiceTest.java

```diff
@@ -399,7 +399,26 @@ public void saveConceptSource_shouldNotSetDateCreatedIfOneIsSuppliedAlready() th
 		
 		Assert.assertEquals(newConceptSource.getDateCreated(), expectedDate);
 	}
-	
+
+	/**
+	 * @see {@link ConceptService#saveDrug(Drug)}
+	 */
+	@Test
+	@Verifies(value = "should pass when drug index is updated on save", method = "saveDrug(Drug)")
+	public void saveDrug_shouldUpdateDrugIndex() throws Exception {
+		Drug drug = new Drug();
+		drug.setName("ASPIRIN BRANDED");
+		drug.setConcept(conceptService.getConceptByName("ASPIRIN"));
+		drug.setStrength("500mg");
+
+		Drug saveDrug = conceptService.saveDrug(drug);
+		assertNotNull(saveDrug);
+
+		List<Drug> drugs = conceptService.getDrugs("ASPIRIN BRANDED");
+		assertEquals(1, drugs.size());
+		assertEquals("ASPIRIN BRANDED", drugs.get(0).getName());
```

Comment:
This test does not prove that updating the index automatically does not work, because the update only takes place when a transaction is committed, which never happens here (the transaction is rolled back by the end of the test). The test would have to be @NotTransactional to prove a bug. Can you try annotating the test with @NotTransactional and running it without your fix in line 301 in saveConcept?